wordpress-mobile/WordPress-Android
# Moves the analytics event names declaration in the Stats enum

## Description
This PR simplifies the `AnalyticsTrackerNosara` by removing switch cases for event names and moving the name declaration in the `Stat` enumeration. It also adds tests verifying that the returned event names did not change.

This would allow for less boilerplate code since in most case adding a new event would only require an addition in the `AnalyticsTracker.Stat` enumeration and optionally setting the event name if that does not match the enum name in lower case. It also helps on avoiding bugs since forgetting to set the event name in the switch statement of the old code did not trigger any compile error.

A next step that I avoided in this PR for simplicity would be to convert the `AnalyticsTrackerNosara` and `AnalyticsTracker` classes to Kotlin.

Internal discussion: p1710265828028039-slack-C04TBSWP09M

-----

## To Test:
* Verify that the tests pass
* Verify that the analytics events are emitted as expected
* Optionally dump all the stats names and verify that nothing changed (see: https://github.com/wordpress-mobile/WordPress-Android/pull/20451#pullrequestreview-1933238170)

-----

## Regression Notes

1. Potential unintended areas of impact

    - Analytics

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    - Added test cases to verify that the event names did not change with this PR

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
